src/server/stat_job.c

```diff
@@ -324,17 +317,15 @@ status_job(job *pjob, struct batch_request *preq, svrattrl *pal, pbs_list_head *
 	/* reset eligible time, it was calctd on the fly, real calctn only when accrue_type changes */
 
 	if (server.sv_attr[(int)SVR_ATR_EligibleTimeEnable].at_val.at_long != 0) {
-		if (get_jattr_long(pjob, JOB_ATR_accrue_type) == JOB_ELIGIBLE) {
+		if (get_jattr_long(pjob, JOB_ATR_accrue_type) == JOB_ELIGIBLE)
 			set_jattr_l_slim(pjob, JOB_ATR_eligible_time, oldtime, SET);
-			pjob->ji_wattr[(int)JOB_ATR_eligible_time].at_flags |= ATR_MOD_MCACHE;
-
-			/* Note: ATR_VFLAG_MODCACHE must be set because of svr_cached() does */
-			/*	 not correctly check ATR_VFLAG_SET */
-		}
 	} else {
 		/* reset the set flags */
-		pjob->ji_wattr[(int)JOB_ATR_eligible_time].at_flags = old_elig_flags;
-		pjob->ji_wattr[(int)JOB_ATR_accrue_type].at_flags = old_atyp_flags;
+		attribute *attr = get_jattr(pjob, JOB_ATR_eligible_time);
+		old_elig_flags = attr->at_flags = old_elig_flags;
+
+		attr = get_jattr(pjob, JOB_ATR_accrue_type);
+		old_atyp_flags = attr->at_flags = old_atyp_flags;
```

Comment:
Not sure I understand what this is doing, can you please clarify? Setting old_atyp_flags changes logic, I think we should avoid that.